[86by7eg12] Fix Tooltip and Informer a11y docs

sheila-semrush opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Motivation and Context

1. Optimized the a11y tab structure and added missing info.
2. Fixed examples to show correct usage for best a11y, removed unnecessary examples.
3. Fixed Informer docs to refer to DescriptionTooltip instead of Hint, updated behaviour description.
4. Edited Tooltip Design tab for better comprehension, clarified differences in behaviour and usage and fixed incorrect info.

Concerns:
1. We recommend merging interactive elements with the Trigger component, but it doesn't work with Button. So the examples currently don't comply with our own recommendations. Will need to look into it together with devs.

## How has this been tested?

Manually.

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] Nice improve.

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly or it's not required.
- [ ] Unit tests are not broken.
- [ ] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [ ] I have added new unit tests on added of fixed functionality.

j-mnizhek commented on website/docs/components/tooltip/static/tooltip-composition.png:
This is a good solution, yep! Let's go back to the roots (we had something similar before all this complex changes to Tooltip)

j-mnizhek commented on website/docs/components/tooltip/static/tooltip-margins.png:
This and the following could be updated, something strange happened to them in Figma 🥲

sheila-semrush replied:
fixed illustrations

j-mnizhek commented on website/docs/components/tooltip/examples/basic_usage.tsx:
Just a thought: maybe we should add color to the `InfoM` and `CheckAltM` icons, so that our users will copy them without forgetting to add color.

j-mnizhek commented on website/docs/components/tooltip/tooltip-api.md:
Suggested some fixes based on the style guide from technical writers (wrote you in DM).

sheila-semrush replied:
linted and fixed

j-mnizhek reviewed (changes requested):
Cool! 😎 Left some small comments here.

And I also still thinking on tooltip-a11y.md, want to propose some changes but need time to sleep on them.

UPD: Discussed them.

sheila-semrush commented:
1. Fixed illustrations.
2. Fixed the language according to the guidelines.
3. Imroved examples (added aria-labels to DescriptionTooltip.Popper, and made the examples a bit more fun).
4. On the Tooltip A11y tab, swapped Considerations and Default attributes places (so Considerations are higher).
5. On the Tooltip A11y tab, reverted DescriptionTooltip attributes to the current values so that docs are in accordance with implementation. Added a note that they're going to change.

j-mnizhek reviewed (approved):
Approve changes 😎 
Everything looks great!

sheila-semrush commented:
Rearranged the example for browser tests to work properly:
<img width="342" alt="image" src="https://github.com/semrush/intergalactic/assets/166654065/3eb1d565-8b5d-4d76-b0eb-1094f3529cf6">
